include/dav1d/headers.rs

```diff
@@ -193,13 +193,38 @@ pub const DAV1D_RESTORATION_WIENER: Dav1dRestorationType =
 pub const DAV1D_RESTORATION_SGRPROJ: Dav1dRestorationType =
     Rav1dRestorationType::SgrProj as Dav1dRestorationType;
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, FromRepr, Default)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, FromRepr, Default)]
+#[repr(u8)]
 pub enum Rav1dRestorationType {
     #[default]
     None = 0,
     Switchable = 1,
     Wiener = 2,
     SgrProj = 3,
+    SgrProj1,
+    SgrProj2,
+    SgrProj3,
+    SgrProj4,
+    SgrProj5,
+    SgrProj6,
+    SgrProj7,
+    SgrProj8,
+    SgrProj9,
+    SgrProj10,
+    SgrProj11,
+    SgrProj12,
+    SgrProj13,
+    SgrProj14,
+    SgrProj15,
+}
+
+impl Rav1dRestorationType {
+    pub fn discriminant(&self) -> u8 {
+        // SAFETY: Because `Self` is marked `repr(u8)`, its layout is a `repr(C)` `union`
+        // between `repr(C)` structs, each of which has the `u8` discriminant as its first
+        // field, so we can read the discriminant without offsetting the pointer.
+        unsafe { *<*const _>::from(self).cast::<u8>() }
```

Comment:
We shouldn't use unsafe here.  You can just do `as u8` since the enum is data-less (only the variants).